src/libraries/System.Private.CoreLib/src/System/Globalization/CultureData.Icu.cs

```diff
@@ -203,6 +203,12 @@ private string IcuGetLocaleInfo(LocaleStringData type, string? uiCultureName = n
             Debug.Assert(!GlobalizationMode.Invariant);
             Debug.Assert(!GlobalizationMode.UseNls);
             Debug.Assert(_sWindowsName != null, "[CultureData.IcuGetLocaleInfo] Expected _sWindowsName to be populated already");
+#if TARGET_BROWSER
+            if (type == LocaleStringData.NativeDisplayName)
```

Comment:
We use the old path, so we fetch data from ICU, because for other types of data it is there.
For `NativeDisplayName` we were till now asking ICU but the ICU loaded to browser did not have it - it returned empty data and we were constructing it on the managed side:
https://github.com/dotnet/runtime/blob/8fbfb3204abba1eb7b42c617a656fa6aff8df84c/src/libraries/System.Private.CoreLib/src/System/Globalization/CultureData.cs#L1086

Reply:
Please add comment about it.